### ISSUE TITLE
Return 404s on invalid bso ids/collection in paths

### ISSUE THREAD

**Issue by pjenvey** [mozilla-services/syncstorage-rs#112]
The Python sets up a regex to validate 'collection' and 'item' (bso ids) within the path:

https://github.com/mozilla-services/server-syncstorage/blob/e992a39/syncstorage/views/__init__.py#L116

This has test_storage::test_handling_of_invalid_bso_fields expecting 404s for these invalid values:

```python
        # Invalid ID - too long
        bso = {"id": "X" * 65, "payload": "testing"}
        ...
        res = self.app.put_json(coll_url + "/" + bso["id"], bso, status=404)
```

Maybe we can emulate this behavior w/ actix's [custom route predicates](https://actix.rs/docs/url-dispatch/#custom-route-predicates)?

**Comment by pjenvey** [comment]
The [fix/112](https://github.com/mozilla-services/syncstorage-rs/compare/fix/112) branch has a fix for this, but due to middleware erroring out (can't parse uid because the path route no longer matched) it doesn't trigger the expected 404s.

Yet another middleware related issue we'll fix later during the actix-web 1.0 migration

**Comment by pjenvey** [comment]
[fix/112](https://github.com/mozilla-services/syncstorage-rs/compare/fix/112) was integrated into #185
